django/db/models.py

```diff
@@ -349,16 +346,16 @@ def __unicode__(self):
 class SmallMolecule(models.Model):
     
     nominal_target          = models.ForeignKey('Protein', null=True)
-    facility_id             = _CHAR(max_length=_FACILITY_ID_LENGTH, **_NOTNULLSTR)
-    salt_id                 = _CHAR(max_length=_SALT_ID_LENGTH, **_NOTNULLSTR)
-    lincs_id                = _CHAR(max_length=15, **_NULLOKSTR)
+    facility_id             = _TEXT(max_length=_FACILITY_ID_LENGTH, **_NOTNULLSTR)
```

Comment:
max_length very much has an effect on our system for **CharField** fields -- it sets the max length of the varchar column. Now for TextField fields, you're right it would be pointless to specify a max_length.